### ISSUE TITLE
Customizable allow/disallow setting for clickable URIs

### ISSUE THREAD

**Issue by MichaelM00** [microsoft/terminal#15570]
# Description of the new feature/enhancement

In the new terminal version it allows all almost all URIs to be clickable as a link but for my use I need file:/// to be the only clickable link type while the other URIs should not be clickable and that they display only as an unsupported link. 

# Proposed technical implementation details (optional)

I rely on the previous terminal method of not generally allowing all URIs but I understand this behavior was changed because some people prefer to make many or all URIs clickable as a link. 

Can there be an option or setting introduced where the user can specify matching only certain URIs such as file:/// as a clickable link and the rest to default to the previous behavior of showing dialog as 'This link type is currently not supported: ... ' if the user has the option configured?

Thank you

**Comment by lhecker** [comment]
Yeah, that generally sounds reasonable to me. But could you clarify _why_ this is important to you? It would help us give this feature request a priority.

**Comment by MichaelM00** [comment]
It is useful for selective viewing of different type of links in output on the terminal. For example I need certain types of links such as file:/// to be clickable but certain other protocol links to not be (they need to be shown only as link plaintext on Ctrl+click). The links in the output are displayed under custom text on terminal so it is currently quite an inconvenience to hover each text, wait and see what type of link each is before deciding whether to Ctrl+click the link or not when my deciding factor is just the protocol/link type.  

Basically, the 'unsupported' links are needed for quick viewing at a glance on the terminal but should not be opened accidentally or clickable (which was the previous behavior). Having the ability for users to specify allowing/disallowing these clickable links for their needs would be great.

**Comment by zadjii-msft** [comment]
maybe this is me just misremembering since I've been out all month, but I thought we only auto-detected file:// and https:// uris. Any other links we make clickable are because the app specifically emitted an OSC8 to make text clickable. Before, we would have still suppressed those, but honestly if the app is asking for clickable text, shouldn't we make it clickable? Like, if the app doesn't want to make text clickable, shouldn't it just _not emit an osc 8_? 

(if I'm totally off base because I cached this out, minimize me as off-topic)

**Comment by j4james** [comment]
My take on this: I like the fact that I can see a link in something like a README file and just Ctrl+Click to open that up in the browser. I trust my browser enough to consider that a reasonably safe action for http(s) links. What I don't trust is the n-billion other protocol handlers that have been registered on the system (here's [one example](https://nakedsecurity.sophos.com/2022/06/02/yet-another-zero-day-sort-of-in-windows-search-url-handling/) why).

If someone decides to embed such a link in a document using `OSC 8`, the only way I'm going to know that it's dodgy is if I hover over every link for several seconds before clicking. That's a massive pain, and a good deal of the time the tooltip doesn't show up anyway. So I'd feel a whole lot safer if I could just disable everything except a handful of trusted protocols.

**Comment by taviso** [comment]
> If someone decides to embed such a link in a document using OSC 8, the only way I'm going to know that it's dodgy is if I hover over every link for several seconds before clicking

But..wt is just using `ShellExecute()`, no?  So an `OSC 8` with `file:///C:/windows/system32/notepad.exe` will just run it?

That seems kinda bad and surprising to me...Not to mention you could obfuscate the link by making it look like it was just auto-detected, but was actually using `OSC 8`. e.g.: `printf '\e]8;;file://c:/windows/system32/notepad.exe\ahttp://safelink/example.pdf\e]8;;\a'`. There are probably sneakier tricks, maybe you could obfuscate the tooltip too?
